### PR TITLE
Running tests with "swift test" on OS X and Linux.

### PR DESCRIPTION
**Pull request by mxl**
- Moved tests to "Tests/CryptoSwiftTests" because `swift test` requires that and now because Tests folder exists `script/CI.sh` will execute `swift test`.
- Added `LinuxMain.swift` and `Error+Extension.swift` for running `swift test` on Linux.
- Changed test methods names to camel case.
- Moved CommonCrypto tests to separate file and excluded it from package.

### CODE REVIEW

File: Tests/CryptoSwiftTests/Access.swift [hunk 1]
```diff
@@ -7,7 +7,8 @@
 //
 
 import XCTest
-import CryptoSwift
+import Foundation
+@testable import CryptoSwift
```
Comment (krzyzanowskim) [inline Tests/CryptoSwiftTests/Access.swift]: the point is to import it without "testable" flag, to test public API

**Comment by krzyzanowskim** [comment]
#218

**Comment by mxl** [comment]
With https://github.com/krzyzanowskim/CryptoSwift/pull/308 fix all tests are passed on both Linux and OS X.

**Comment by mxl** [comment]
Is there anything else I could help with?

**Comment by krzyzanowskim** [comment]
I really like it, and I'm happy to merge this one. Thank you!

PS. I was just busy.

**Pull request merged**